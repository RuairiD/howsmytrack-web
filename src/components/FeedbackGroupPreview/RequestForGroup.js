import React from "react";

import { Row, Col, Typography } from "antd";

import MediaEmbed from "../MediaEmbed/MediaEmbed";

const RequestForGroup = ({ mediaUrl, mediaType }) => {
    if (mediaUrl) {
        return (
            <Row gutter={[16, 16]}>
                <Col>
                    <Typography.Text strong>You submitted:</Typography.Text>
                    <MediaEmbed
                        mediaUrl={mediaUrl}
                        mediaType={mediaType}
                        size="small"
                    />
                </Col>
            </Row>
        );
    }
    return (
        <Row gutter={[16, 16]}>
            <Col>
                <Typography.Text strong>You did not submit a track for this group.</Typography.Text>
            </Col>
        </Row>
    );
};

export default RequestForGroup;
