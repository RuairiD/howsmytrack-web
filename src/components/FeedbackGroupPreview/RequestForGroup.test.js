import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RequestForGroup from "./RequestForGroup";

describe("RequestForGroup", () => {
    it("renders an inline feedback request media preview", () => {
        const wrapper = shallow(
            <RequestForGroup
                mediaUrl="mediaUrl"
                mediaType="mediaType"
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a trackless inline feedback request preview", () => {
        const wrapper = shallow(
            <RequestForGroup />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
