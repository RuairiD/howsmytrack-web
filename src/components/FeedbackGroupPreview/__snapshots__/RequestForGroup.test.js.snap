// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForGroup renders a trackless inline feedback request preview 1`] = `
<Row
  gutter={
    Array [
      16,
      16,
    ]
  }
>
  <Col>
    <Text
      strong={true}
    >
      You did not submit a track for this group.
    </Text>
  </Col>
</Row>
`;

exports[`RequestForGroup renders an inline feedback request media preview 1`] = `
<Row
  gutter={
    Array [
      16,
      16,
    ]
  }
>
  <Col>
    <Text
      strong={true}
    >
      You submitted:
    </Text>
    <MediaEmbed
      mediaType="mediaType"
      mediaUrl="mediaUrl"
      size="small"
    />
  </Col>
</Row>
`;
